### PR TITLE
Add metrics for garbage collection

### PR DESCRIPTION
**Pull request by congqixia**
See also #27202
/kind improvement

### CODE REVIEW

**Comment by congqixia** [comment]
run 2 failed due to create index timeout

**Comment by congqixia** [comment]
run 3 failed due to deployment failure

File: pkg/metrics/datacoord_metrics.go [hunk 1]
```diff
@@ -149,6 +149,26 @@ var (
...
+	/* garbage collector related metrics */
+
+	// GarbageCollectorListLatency metrics for gc scan storage files.
+	GarbageCollectorListLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataCoordRole,
+			Name:      "gc_list_latency",
+			Help:      "latency of list objects in storage while garbage collecting (in seconds)",
+			Buckets:   []float64{0.001, 0.1, 0.5, 1, 5, 10, 20, 50, 100, 250, 500, 1000, 3600, 5000, 10000}, // unit seconds
```
Comment (LoveEachDay) [inline pkg/metrics/datacoord_metrics.go:161]: We have a common aggreement that histogram buckets in milliseconds unit.

File: internal/datacoord/garbage_collector.go [hunk 2]
```diff
@@ -153,6 +157,7 @@ func (gc *garbageCollector) scan() {
 			)
 		}
 		segmentMap, filesMap := getMetaMap()
+		metrics.GarbageCollectorListLatency.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), labels[idx])
```
Comment (LoveEachDay) [inline internal/datacoord/garbage_collector.go:160]: Here we'd call the `#Observe(duration)` to add a observe.

File: pkg/metrics/datacoord_metrics.go (hunk 1 above)
Comment (congqixia) [reply pkg/metrics/datacoord_metrics.go:161]: this common bucket is too small for gc duration

File: internal/datacoord/garbage_collector.go (hunk 2 above)
Comment (congqixia) [reply internal/datacoord/garbage_collector.go:160]: my fault, fixing it

**Pull request merged**